services/meta/client.go

```diff
@@ -305,7 +305,7 @@ func (c *Client) Database(name string) (*DatabaseInfo, error) {
 		}
 	}
 
-	return nil, nil
+	return nil, fmt.Errorf("database '%s' does not exist", name)
```

Comment:
I'm wondering if we should make this an exported error (like these: https://github.com/influxdata/influxdb/blob/master/services/meta/errors.go#L38) in case the caller ever needs to determine what type of error it got.  Detecting a `Not Found` style error is pretty common.

Although, fwiw, I always preferred the style of error checking like the [os](https://golang.org/pkg/os/#IsNotExist) package does for `IsExistError`.  However, that isn't a convention that we use in the current code base.  I like it mostly as it doesn't require us to expose errors as part of our API.